### ISSUE TITLE
cabal version should be included as an requirement in somewhere

### ISSUE THREAD

**Issue by jonaprieto** [asr/apia#88]
I got this yesterday in a clean installation of fedora 24.

``` bash
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.8.4
$ cabal --version
cabal-install version 1.18.1.0
using version 1.18.1.5 of the Cabal library 
$ agda --version
Agda version 2.6.0.1-5893007
$ pwd
/home/hotel/apia
$ make install-bin
cabal install --disable-documentation
Resolving dependencies...
Configuring apia-1.0.1...
Failed to install apia-1.0.1
Last 10 lines of the build log ( /home/hotel/.cabal/logs/apia-1.0.1.log ):
cabal: Error: some packages failed to install:
apia-1.0.1 failed during the configure step. The exception was:
user error (
Setup.hs:25:5:
Module
‘Distribution.Package’
does not export
‘PackageName(PackageName, unPackageName)’
)
Makefile:490: recipe for target 'install-bin' failed
make: *** [install-bin] Error 1
``` 

After upgrade cabal to the version 1.24, the issue above disappeared.

**Comment by asr** [comment]
I'll fix it.

**Comment by jonaprieto** [comment]
What about adding the version for cabal-install in the apia.cabal?

**Comment by asr** [comment]
No. We need to added a dependence on the Cabal library (no cabal-install).

**Comment by asr** [comment]
Travis jobs with GHC 7.8.4 and 7.6.3 are failing.

**Comment by asr** [comment]
The revert closed the issue.

**Comment by asr** [comment]
The previous fix was wrong because `Setup.hs` doesn't use the dependencies in `apia.cabal`. A proper fix could use CPP in `Setup.hs`but it is not working and I'm testing...

**Comment by jonaprieto** [comment]
I see.

**Comment by asr** [comment]
> A proper fix could use CPP in `Setup.hs`but it is not working and I'm testing...

CPP in `Setup.hs` works with GHC 8.0.2 but doesn't work with GHC 7.8.4. I'll try a different solution...

**Comment by asr** [comment]
> I'll try a different solution...

First step: see dca667d15b60abf78b8979b00f955469b5ec351b.

**Comment by asr** [comment]
Fixed by dca667d15b60abf78b8979b00f955469b5ec351b.
